### DISCUSSION TITLE
Semantic release Action fails with git plugin

### DISCUSSION THREAD

**Discussion by lethalbrains** [semantic-release/semantic-release#2032] [Configuration]
**Describe the bug**
Semantic release was working perfectly with  github actions but started failing since today. There hasn't been any changes done to the code

**release.config.js **

    module.exports = {
        branches: [
            {name: 'main'}, 
    		{name: 'beta', prerelease: true}
        ],
    	plugins: [
            "@semantic-release/commit-analyzer",
            "@semantic-release/release-notes-generator",
            "@semantic-release/git",
            "@semantic-release/github",
    		[
    			'@semantic-release/npm',
    			{
    				npmPublish: false
    			}
    		]
    	]
    };

**action.yml**

    name: Deployment
    
    on:
      push:
        branches:
          - main
          - beta
    
        tags:
          - v*
    
      pull_request:
    
    jobs:
    
      build-release: 
        runs-on: ubuntu-latest
        steps:
          - name: Checkout
            uses: actions/checkout@v2
          - name: Semantic Release
            id: semantic
            uses: cycjimmy/semantic-release-action@v2
            env:
              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

**Expected behavior**
An update in the version with every git push.

**Current behavior**
![Error](https://i.imgur.com/Zu3bBGO.png "Error Message")

**Additional context**
The issue is occurs only when I add `@semantic-release/git` to the `release.config.js`. When its removed, everything works as expected.

**Comment by travi** [comment]
> Semantic release was working perfectly with github actions but started failing since today. There hasn't been any changes done to the code

we do not maintain the `cycjimmy/semantic-release-action` action. if something changed in the behavior of that action, you need to request support from the maintainers of that action

> The issue is occurs only when I add `@semantic-release/git` to the `release.config.js`. When its removed, everything works as expected.

unless i'm misunderstanding you, this contradicts your statement that nothing changed. do you have the git plugin package installed? semantic-release does not automatically install plugins based on the config file

**Reply by lethalbrains** [reply]
@travi , I had git plugin in release.config.js and added it as an extra plugin too. I am getting the above mentioned error if I have git plugin in both places or even just in the `release.config.js`(as shown in the actual issue). When I remove the git plugin from both places, github actions was able to trigger semantic release.  

      with:
        extra_plugins: |
          @semantic-release/git

**Reply by awalias** [reply]
we're also having some issues with [releases not triggering](https://github.com/supabase/gotrue/runs/3101524626?check_suite_focus=true#step:5:31) in the last few days with:
```
      - name: Release on GitHub
        id: github-release
        run: npx semantic-release -p        \
          @semantic-release/commit-analyzer \
          @semantic-release/github         \
          @semantic-release/release-notes-generator
```

[action here](https://github.com/supabase/gotrue/blob/master/.github/workflows/release.yml)

**Reply by travi** [reply]
@awalias the log that you linked to clearly shows that the two commits being analyzed are `chore` type and should not trigger a release:

```
[7:18:07 AM] [semantic-release] › ℹ  Found 2 commits since last release
[7:18:07 AM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[7:18:07 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Merge pull request #144 from supabase/chore/fix-lint

chore: fix incorrect struct tag
[7:18:07 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[7:18:07 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: fix incorrect struct tag
[7:18:07 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[7:18:07 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analysis of 2 commits complete: no release
```

please open a new discussion for issues unrelated to an an existing thread.

**Reply by travi** [reply]
> I had git plugin in release.config.js and added it as an extra plugin too

@lethalbrains i don't think that answers my question. the log suggests that you do not have the git plugin installed. you have not clarified to me whether you have taken a step to install the extra dependency. are you suggesting that `extra_plugins` is intended to do that? if so, that is a feature of the action that you are using, not of `semantic-release` itself.


as stated earlier:

> we do not maintain the cycjimmy/semantic-release-action action. if something changed in the behavior of that action, you need to request support from the maintainers of that action

if you think you have configured that action in a way that should work, you need to seek support from the maintainers of that action. we dont have any inside information about how it sets up its environment beyond the public documentation.
